servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -2638,6 +2655,48 @@ public DomainMetaStoreValidValuesList getDomainMetaStoreValidValuesList(Resource
         return domainMetaStoreValidValuesList;
     }
 
+    @Override
+    public AuthHistoryList getAuthHistoryList(ResourceContext ctx, String domainName) {
+        final String caller = ctx.getApiName();
+        logPrincipal(ctx);
+
+        validateRequest(ctx.request(), caller);
+        validate(domainName, TYPE_DOMAIN_NAME, caller);
+
+        // for consistent handling of all requests, we're going to convert
+        // all incoming object values into lower case (e.g. domain, role,
+        // policy, service, etc name)
+
+        domainName = domainName.toLowerCase();
+        setRequestDomain(ctx, domainName);
+
+        List<AuthHistoryRecord> authHistoryRecords = dbService.getAuthHistory(domainName);
+        List<AuthHistory> authHistoryList = authHistoryRecords.stream()
+                .map(record -> {
+                    AuthHistory authHistory = new AuthHistory();
+                    authHistory.setDomainName(record.getDomain());
+                    authHistory.setPrincipal(record.getPrincipal());
+                    authHistory.setEndpoint(record.getEndpoint());
+                    authHistory.setTimestamp(getAuthHistoryTimeStamp(record, caller));
+                    authHistory.setTtl(record.getTtl());
+                    return authHistory;
+                })
+                .collect(Collectors.toList());
+        return new AuthHistoryList().setAuthHistoryList(authHistoryList);
+    }
+
+    private Timestamp getAuthHistoryTimeStamp(AuthHistoryRecord authHistoryRecord, String caller) {
+        try {
+            SimpleDateFormat formatter = new SimpleDateFormat("dd/MMM/yyyy:HH:mm:ss");
+            Date date = formatter.parse(authHistoryRecord.getTimestamp());
```

Comment:
what format are we storing the timestamps in the store? can't we just assume the use of unix epoch time so we don't have to deal with formatting and parsing of these values?

Reply:
During sync from Access logs to DynamoDB, I'm extracting the time directly from the log and push it as is to DynamoDB (the format is dd/MMM/yyyy:HH:mm:ss).
The reason is that I expect the syncer to be in heavier load during the sync than Athenz when a user fetches these records for a specific domain. So I prefer the parsing to take place once we pull the records instead of doing it when the records are pushed to the DynamoDB.
Let me know if you think I should change the date to UNIX epoch time during push.